scaffolding-ruby/doc/reference.md

```diff
@@ -0,0 +1,473 @@
+* If the `nokogiri` gem is detected, then XML/XSLST-related Habitat packages will be injected into your Plan's `pkg_deps` array.
+* If the `execjs` gem is detected, then Node-related Habitat packages will be injected into your Plan's `pkg_deps` array.
+* If the `webpacker` gem is detected, then Yarn-related Habitat packages will be injected into your Plan's `pkg_deps` array.
+
+Additional checks performed by this scaffolding are:
+
+* The app's version of Ruby will be determined by checking several source locations. See the Ruby Version section for more details.
+* If your app's root directory contains a `.git/` subdirectory, then Git-related Habitat packages will be injected into your Plan's `pkg_build_deps` array to better support installing your app while honoring the `.gitignore` file.
+
+###  Specifying Run Dependencies in Your Plan
+
+Generally speaking, your app needs additional Habitat packages present at runtime for one of two reasons:
+
+1. Your app expects certain programs present which it may shell out  or bind to in some way.
+2. Your `Gemfile` has gem dependencies with native extensions, requiring additional linked libraries.
+
+In both cases, this means your app needs additional Habitat package installed at runtime. For this, we use the `pkg_deps` array in your Plan.
+
+For this example, let's suppose you are using the [rbnacl][] gem for bindings to the [libsodium][] library. Additionally, your app resizes images to generate thumbnails by using the `mogrify` program from the [ImageMagick][] software. To add both of these packages, update your `pkg_deps` line to the following below:
+
+```sh
+pkg_name=my_app
+pkg_origin=acmecorp
+pkg_version=0.1.0
+ # ...
+pkg_scaffolding=core/scaffolding-ruby
+pkg_deps=(core/imagemagick core/libsodium)
+```
+
+###  Specifying Build Dependencies in Your Plan
+
+Finally, your app may require additional software present in order to **build** your app that is not present by default.
+
+For this example, let's suppose you are using the [Helix][] project to write some Ruby classes in Rust. For this you need the Rust compiler present, but only at build time. To add the Rust compiler package, update your `pkg_build_deps` line to the following below:
+
+```sh
+pkg_name=my_app
+pkg_origin=acmecorp
+pkg_version=0.1.0
+ # ...
+pkg_scaffolding=core/scaffolding-ruby
+pkg_build_deps=(core/rust)
+```
+
+## Ruby Version
+
+### Selecting a Version of Ruby
+
+By default the latest version of the [MRI][]-based [`core/ruby`][] package will be injected into your Plan's `pkg_deps` array. To specify a non-default version of Ruby, there are two locations you can do this:
+
+1. Use the [`ruby`][gemfile_ruby] keyword in your app's `Gemfile`
+2. Set the `scaffolding_ruby_pkg` variable in your Plan with a valid Habitat package identifier corresponding to a package with a `ruby` program
```

Comment:
I wouldn't worry to much about the length, as you've already realized these just need to read well and standout. We provide autocompletion plugins to editors and people can just copy paste the word `scaffolding_` if they need to set a number of these